staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go

```diff
@@ -179,6 +179,11 @@ func (u *Unstructured) GetOwnerReferences() []metav1.OwnerReference {
 }
 
 func (u *Unstructured) SetOwnerReferences(references []metav1.OwnerReference) {
+	if references == nil {
+		RemoveNestedField(u.Object, "metadata", "ownerReferences")
```

Comment:
hi @nikhita, how about having an additional check to respect the field tag `omitempty` via `reflect`? we might only do the removal when it's tagged, amiright?

Reply:
it's working tho, but a dynamic approach would be better for maintenance

Reply:
Reflection is slow to use in the normal path. A reflection based unit test to catch drift might be better.

Reply:
> Reflection is slow to use in the normal path.

true. it's a heavy cost to do the reflection every time accessing normal paths. or we could simply have a thread-unsafe map like `map[(path)string](isOmitemptyField)bool` and put all things we need inside `init()` call